libs/platform/src/lib/components/form/checkbox/checkbox.component.ts

```diff
@@ -0,0 +1,104 @@
+import { NgControl, NgForm } from '@angular/forms';
+import { Component, ChangeDetectorRef, Self, Optional, Input, ViewChild, AfterViewInit } from '@angular/core';
+import { CheckboxComponent as CoreCheckboxComponent, stateType } from '@fundamental-ngx/core';
+import { BaseInput } from '../base.input';
+import { Status } from './../form-control';
+
+let uniqueId = 0;
+
+@Component({
+    selector: 'fdp-checkbox',
+    templateUrl: './checkbox.component.html',
+})
+export class CheckboxComponent extends BaseInput implements AfterViewInit {
+    /** reference of child checkbox implementation */
+    @ViewChild(CoreCheckboxComponent)
+    corecheckbox: CoreCheckboxComponent;
+
+    /** set to true if binary checkbox */
+    @Input()
+    binary: boolean = false;
+
+    /** Sets label for checkbox. */
+    @Input()
+    label: string;
+
+    /** indeterminate state */
+    @Input()
+    tristate: boolean = false;
+
+    /** indeterminate state can be selected  */
+    @Input()
+    tristateSelectable: boolean = true;
+
+    /** value for checkbox, when it is not binary */
+    @Input()
+    get value(): any {
+        return super.getValue();
+    }
+    set value(value: any) {
+        if (value) {
+            this._value = value;
+        }
+    }
+
+    /** set status value , will be coming from checkbox group*/
+    set stateStatus(newStatus: Status) {
+        this._status = newStatus;
+        this._cd.markForCheck();
+    }
+
+    /** returns form status */
+    get stateStatus(): Status {
+        return this._status;
+    }
+
+    /** @hidden captures controller value */
+    private modelValue: any;
+
+    /** tracking checkbox current value */
+    checkboxValue: any;
```

Comment:
another one, this is too complicated.  The checkbox is capable of setting value directly.

Reply:
This is needed to notify corecheckbox for current checkbox value. based on this value checkbox state changes.

Reply:
Ok. Let's pass this. I think we can use single _value